src/main/java/com/imgix/URLHelper.java

```diff
@@ -158,6 +161,58 @@ private static String joinList(List<String> strings, String separator) {
     return sb.toString();
   }
 
+  // checkProxyStatus checks if the path has one of the four possible
+  // acceptable proxy prefixes. First we check if the path has the
+  // correct ascii prefix. If it does then we know that it is a proxy,
+  // but it's not percent encoded. Second, we check if the path is
+  // prefixed by a percent-encoded prefix. If it is, we know that it's
+  // a proxy and that it's percent-encoded. Finally, if the path isn't
+  // prefixed by any of these four prefixes, it is not a valid proxy.
+  // This might be "just enough validation," but if we run into issues
+  // we can make this check smarter/more-robust.
+  public static Map<String, Boolean> checkProxyStatus(String p) {
+    Map<String, Boolean> status = new HashMap<String, Boolean>();
+    String path = p;
+    path.replaceAll("^/", "");
```

Comment:
Had to change implementation to HashMap since JDK 8 was not very happy with `Map`. Makes syntax read a little differently here than in does in go.